### DISCUSSION TITLE
How do you configure Zed to avoid moving the cursor to the first search result minibuffer?

### DISCUSSION THREAD

**Discussion by anthonyryan1** [zed-industries/zed#17837] [Feature Requests]
When searching all files in a project, the cursor moves out of the search box and into the cursor position of the first minibuffer.

How do I configure Zed to keep my cursor in the search box until I move it manually?

I often increment on my searches, adding or removing a few characters and hitting enter. The fact that the cursor jumps to editing the first result after every enter key violates my expectations, because I expect to be still in the search box (I never clicked or tabbed out of it).

**Comment by vidjul** [comment]
Any updates on this? :pray:

**Comment by failable** [comment]
Same question, since Zed has no incremental search, this behavior is super annoying.

**Comment by lexeech** [comment]
Absolutely the same question.
Let me decide if I want to focus on the first result after pressing Enter in Project Search or keep focus on input.

**Comment by iyht** [comment]
In case you are using vim mode, I just do `g/` to get back to the minibuffer to start typing in the minibuffer again.

**Reply by lexeech** [reply]
Not my case, I have to redo - `cmd+shift+f` each time I want to adjust my search which is quite frustrating ...

**Comment by hmasovic** [comment]
@as-cii @ConradIrwin any update on this?

**Comment by lexeech** [comment]
Same question. Very annoying "feature".

**Comment by ConradIrwin** [comment]
Weird that this is different in project search vs. buffer search. Happy to pair with someone on a fix: https://cal.com/conradirwin/pairing

**Reply by hmasovic** [reply]
as a side effect, doing `cmd-shift-f`, after the initial search, copies the value where the first cursor lies, so if you try to do `cmd-shift-f` again (because the cursor went down), it can tend to overwrite the initial value

**Reply by hmasovic** [reply]
That's under the case it doesn't find the exact value being searched, if it finds the exact same searched input - then it does a c/p from the looks of it

**Reply by hmasovic** [reply]
@ConradIrwin do you need someone with rust experience?

**Reply by ConradIrwin** [reply]
Nope! Happy to provide the rust experience :D

**Reply by lexeech** [reply]
Hey guys, any updates?

**Comment by hasparus** [comment]
It would be amazing to be able to change this.

**Reply by lexeech** [reply]
Idk I feel like I better learn vim and set it up how I like rather than wait for this setting ...

**Comment by iyht** [comment]
Hi guys. I just created this PR(https://github.com/zed-industries/zed/pull/38450) for implementing incremental project search. I think it should solve this problem. Let me know what you guys think!

https://github.com/user-attachments/assets/062d244c-2471-421c-8420-56a941c70af1

**Comment by brendanmc6** [comment]
Would love an update on this! Quite a destructive behavior, and an extremely difficult habit to break coming from VS Code... Pressing Enter after a search-all (ctrl+shift+a) should not relocate the cursor. This is not how any search input field has ever worked that I'm aware of...

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
